pkg/util/common.go

```diff
@@ -988,8 +988,8 @@ func GetRouteClient(restConfig *rest.Config) *routeclient.RouteV1Client {
 		log.Errorf("unable to get route client")
 		return nil
 	}
-	_, err = GetRoute(routeClient, "default", "docker-registry")
-	if err != nil && strings.Contains(err.Error(), "could not find the requested resource") && strings.Contains(err.Error(), "openshift.io") {
+	_, err = ListRoutes(routeClient, "default", "")
+	if err != nil && (strings.Contains(err.Error(), "could not find the requested resource") || strings.Contains(err.Error(), "cannot list routes")) && strings.Contains(err.Error(), "openshift.io") {
```

Comment:
Let's not look for fixed strings.  That's very fragile.  If there's an error getting routes then the route client is disabled.  Is there a need for more complexity?

Reply:
For get, we need this complexity bcoz of openshift versions.. But for list, you are right, we don't need to compare